### PR TITLE
add RouterEthHlpBalancer

### PR DESCRIPTION
**Pull request by Preston-Harrison**
Adds a route from ETH => Balancer, and vice versa by using the HlpBalancer contract and WETH

### CODE REVIEW

**Review by bitcoinbrisbane — DISMISSED** [review]

File: contracts/RouterEthHlpBalancer.sol [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+            weth,
+            poolId,
+            amountIn,
+            minOut,
+            _self,
+            signedQuoteData
+        );
+
+        _unwrapWETH(weth);
+        payable(receiver).transfer(_self.balance);
```
Comment (rellfy) [inline contracts/RouterEthHlpBalancer.sol:64]: Aren't we missing a `minOut` check?

File: contracts/RouterEthHlpBalancer.sol [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            weth,
+            hlpBalancerToken,
+            tokenOut,
+            poolId,
+            msg.value,
+            minOut,
+            receiver,
+            signedQuoteData
+        );
+    }
```
Comment (rellfy) [inline contracts/RouterEthHlpBalancer.sol:91]: We should add a minOut check here too

File: contracts/RouterEthHlpBalancer.sol [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+    receive() external payable {}
```
Comment (rellfy) [inline contracts/RouterEthHlpBalancer.sol:93]: Why?

Comment (Preston-Harrison) [reply contracts/RouterEthHlpBalancer.sol:93]: otherwise the contract cannot receive eth

File: contracts/RouterEthHlpBalancer.sol (hunk 1 above)
Comment (Preston-Harrison) [reply contracts/RouterEthHlpBalancer.sol:64]: the routerHlpBalancer already does this, but I'll add a second one

**Review by rellfy — APPROVED** [review]

**Review by rellfy — APPROVED** [review]

**Review by nabsta-eth — APPROVED** [review]

**Pull request merged**